packages/core/src/dev/activation.ts

```diff
@@ -204,10 +211,21 @@ class VirtualObjectFile implements FileProvider {
 
     public constructor(private readonly storage: vscode.Memento | vscode.SecretStorage, private readonly key: string) {}
 
+    /** Emits an event indicating this file's content has changed */
+    public refresh() {
+        /**
+         * Per {@link vscode.FileSystemProvider.onDidChangeFile}, if the mTime does not change, new file content may
+         * not be retrieved. Without this, when we emit a change the text editor did not update.
+         */
+        this.mTime++
+        this.onDidChangeEmitter.fire()
+    }
+
     public stat(): { ctime: number; mtime: number; size: number } {
         // This would need to be filled out to track conflicts
-        return { ctime: 0, mtime: 0, size: 0 }
+        return { ctime: 0, mtime: this.mTime, size: 0 }
     }
+    private mTime = 0
```

Comment:
nice approach. nit: should this variable live near the other object state (line 210)